lib/oppen/token.rb

```diff
@@ -108,12 +148,41 @@ def break_type_name
       end
     end
 
-    # End Token
+    # End Token.
     class End < Token
       nil
     end
 
-    # EOF Token
+    # The EOF token is a token present in Oppen's original paper and algorithm.
+    # It can be interpreted as a flush of the output.
+    # Multiple EOF tokens can be present in the same list of tokens.
+    #
+    # @example
+    #   tokens = [
+    #     Oppen::Token::Begin.new,
+    #     Oppen::Token::String.new('XXXXXXXXXX'),
+    #     Oppen::Token::End.new,
+    #     Oppen::Token::EOF.new,
+    #     Oppen::Token::Begin.new,
+    #     Oppen::Token::String.new('YYYYYYYYYY'),
+    #     Oppen::Token::End.new,
+    #   ]
+    #   Oppen.print tokens:
+    #
+    #   => XXXXXXXXXX
+    #
+    #   tokens = [
+    #     Oppen::Token::Begin.new,
+    #     Oppen::Token::String.new('XXXXXXXXXX'),
+    #     Oppen::Token::End.new,
+    #     Oppen::Token::Begin.new,
+    #     Oppen::Token::String.new('YYYYYYYYYY'),
+    #     Oppen::Token::End.new,
+    #     Oppen::Token::EOF.new,
+    #   ]
+    #   Oppen.print tokens:
+    #
+    #   => XXXXXXXXXXYYYYYYYYYY
     class EOF < Token
```

Comment:
We can use comments:

```suggestion
    #   tokens = [
    #     Oppen::Token::Begin.new,
    #     Oppen::Token::String.new('XXXXXXXXXX'),
    #     Oppen::Token::End.new,
    #     Oppen::Token::EOF.new,
    #     Oppen::Token::Begin.new,
    #     Oppen::Token::String.new('YYYYYYYYYY'),
    #     Oppen::Token::End.new,
    #   ]
    #   puts Oppen.print(tokens)
    #   # Prints the following:
    #   #
    #   # XXXXXXXXXX
    #
    #   tokens = [
    #     Oppen::Token::Begin.new,
    #     Oppen::Token::String.new('XXXXXXXXXX'),
    #     Oppen::Token::End.new,
    #     Oppen::Token::Begin.new,
    #     Oppen::Token::String.new('YYYYYYYYYY'),
    #     Oppen::Token::End.new,
    #     Oppen::Token::EOF.new,
    #   ]
    #   puts Oppen.print(tokens)
    #   # Print the following:
    #   #
    #   # XXXXXXXXXXYYYYYYYYYY
    #
    class EOF < Token
```